docs/DeveloperGuide.md

```diff
@@ -1,15 +1,596 @@
+such as getting, adding, editing, deleting and purging of transaction(s) in the list.
+
+The `Transaction` class is the abstract classes of an `Income` or an `Expense`. A more detailed explanation on the 
+implementation on the transactions can be viewed under Section
+[Implementation for Transaction](#implementation-for-transaction).
+
+#### How the data component interacts
+
+- When MoolahManager starts running, the `Duke` class will initialize a `Storage` object which will attempt to 
+read from the file and initialize both `budget` and `transactionList`. The temporary `transactionList` containing all the stored 
+transaction records will be returned by the `Storage`. 
+Based on the whether the initialization is successful, the corresponding constructor will be called to initialize a 
+`transactionList` object which will be used throughout the application running time to hold the `transactions` added.
+
+  ![Sequence Diagram on Creation of TransactionList](images/TransactionListSequenceDiagram.png)
+
+- A transaction (either an income or expense) is created by an `addCommand` class, can be modified by an `editCommand` 
+class and can be deleted by a `deleteCommand` or `purgeCommand` class. These interactions are described in further detail
+under each command section below.
+
+- The monthly budget can be updated by `budget` command.
+
+<!-- TODO: Describe how category and categoryList work here -->
+
+_Written by: Chia Thin Hong_
+
+### Storage Component
+The `Storage` component is a standalone class. It utilises its sub-methods and methods from external classes to perform it's read and write functions.
+
+The structure of `Storage` can be seen below.
+<p align="center">
+    <img src="images/StorageComponentClassDiagram.png">
+    <br />
+    <i> Simplified Class Diagram for Storage Component</i>
+</p>
+
+1. `Duke` initializes `Storage` and `Storage#initializeFile` is called.
+2. During the initialization , parser methods from `CommandParser` and `ParameterParser` would be used to process the entries within `Duke.txt`.
+Methods from `Budget` and `TransactionList` would be used for the storage of `Budget` amount and `TransactionList` entries into the program.
+3. `TransactionList` is returned to `Duke` after the storage of entries within `Duke.txt`.
+4. After initialization and upon user input, `Command` classes such as `AddCommand`can call for `Storage#writeToFile` method in order to update the contents within `Duke.txt`.
+
+_Written by: Yong Chin Han_
+
+### Parser Component
+The Parser component comprises of two main parsers: `CommandParser` and `ParameterParser`. Together, both these 
+parsers are used to generate a command object with its accurate parameters according to the input from the UI. 
+
+The structure of the data component in Moolah Manager is illustrated in the class diagram below:
+![Data Component Class Diagram](images/ParserClassDiagram.png)
```

Comment:
Do consider removing some functions in the ParameterParser class if they are not used to describe how the component works.